packages/loopring_v3/contracts/core/impl/ExchangeV3.sol

```diff
@@ -441,7 +441,6 @@ contract ExchangeV3 is IExchangeV3
         uint96  amount,
         address feeToken,
         uint96  fee,
-        uint    data,
```

Comment:
The operator can create more than one transfer with the same parameters but just different "data", so maybe we should include this field. Otherwise, we can simply just use {from, token, storageID} as the unique transfer identifier.

Reply:
See above, needs to contain all the transfer parameters, the approval needs to approve all the data of the transfer. `storageID` is used to make the transfer unique when all else is the same.

Reply:
I agree, that's why I think `data` should not be removed from this method.

Reply:
I'm not sure I see how `data` helps. `storageID` is the important one here because once the `storageID` is used none of the other transactions with that `storageID` can be executed afterwards. So in theory you could use `data` to make it unique, but only one will be able to be executed. Is that what you want? Using `validUntil` for that would also work then I think.